From e474ca74f56c2bd39f1e79d254f67a5383246fba Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@foss.st.com>
Date: Mon, 18 Mar 2024 14:08:20 +0100
Subject: [PATCH 3/4] tests/modetest: set property in atomic mode

In atomic mode, the additional properties must be done after
call of functions atomic_set_mode & atomic_set_planes.

Signed-off-by: Yannick Fertre <yannick.fertre@st.com>
Upstream-Status: Inappropriate
---
 tests/modetest/modetest.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/modetest/modetest.c b/tests/modetest/modetest.c
index 0fb5f21..dad9316 100644
--- a/tests/modetest/modetest.c
+++ b/tests/modetest/modetest.c
@@ -2463,6 +2463,9 @@ int main(int argc, char **argv)
 
 			if (count)
 				atomic_clear_mode(&dev, pipe_args, count);
+
+			for (i = 0; i < prop_count; ++i)
+				set_property(&dev, &prop_args[i]);
 		}
 
 		ret = drmModeAtomicCommit(dev.fd, dev.req, DRM_MODE_ATOMIC_ALLOW_MODESET, NULL);
@@ -2474,6 +2477,9 @@ int main(int argc, char **argv)
 
 		drmModeAtomicFree(dev.req);
 	} else {
+		for (i = 0; i < prop_count; ++i)
+			set_property(&dev, &prop_args[i]);
+
 		if (dump_path) {
 			fprintf(stderr, "writeback / dump is only supported in atomic mode\n");
 			return 1;
-- 
2.34.1

